client/app/pages/queries/view.js

```diff
@@ -43,26 +43,31 @@ function QueryViewCtrl($scope, Events, $route, $routeParams, $http, $location, $
     return dataSourceId;
   }
 
-  function updateSchema() {
-    $scope.hasSchema = false;
-    $scope.editorSize = 'col-md-12';
-    DataSource.getSchema({ id: $scope.query.data_source_id }, (data) => {
-      if (data && data.length > 0) {
+  function toggleSchemaBrowser(hasSchema) {
+    $scope.hasSchema = hasSchema;
+    $scope.editorSize = hasSchema ? 'col-md-9' : 'col-md-12';
+  }
+
+  function getSchema(refresh = undefined) {
+    DataSource.getSchema({ id: $scope.query.data_source_id, refresh }, (data) => {
+      const hasSchema = data && (data.length > 0);
+      if (hasSchema) {
```

Comment:
Yes, it would be great!
Which HTTP code we'll get when error occurs? Will it be 5xx or just 200 with empty data?